cmd/add.go

```diff
@@ -69,12 +69,17 @@ if [ -t 1 ] ; then
   exec < /dev/tty ; # <- enables interactive shell
 fi
 
+dir="$(cd "$(dirname $(dirname $(dirname "${BASH_SOURCE[0]}")))" >/dev/null 2>&1 && pwd)"
+
 ` + autoInstall(hookName, fs) + "\n" + "cmd=\"lefthook run " + hookName + " $@\"" +
 		`
 
 if lefthook -h >/dev/null 2>&1
 then
   eval $cmd
+elif test -f "$dir/node_modules/@arkweid/lefthook/bin/lefthook"
+then
+  eval $dir/node_modules/@arkweid/lefthook/bin/$cmd
```

Comment:
You can see we're checking if the binaries in the NPM packages exists **before** the Ruby part. This seems a bit odd at the first glance… but it improves performance for NPM packages because running `bundle exec` is quite slow. And adding a simple file existence check before the Ruby part is not an issue: the check is fast enough to not penalize Ruby users.